UtilsDynamicSimulations/OpenSimAD/utilsKineticsOpenSimAD.py

```diff
@@ -0,0 +1,420 @@
+'''
+    ---------------------------------------------------------------------------
+    OpenCap processing: utilsKineticsOpenSimAD.py
+    ---------------------------------------------------------------------------
+
+    Copyright 2023 Stanford University and the Authors
+    
+    Author(s): Antoine Falisse, Scott Uhlrich
+    
+    Licensed under the Apache License, Version 2.0 (the "License"); you may not
+    use this file except in compliance with the License. You may obtain a copy
+    of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+'''
+
+import os
+import numpy as np
+import pandas as pd
+import utils
+import opensim
+
+class kineticsOpenSimAD:
+    
+    def __init__(self, data_dir, session_id, trial_name, case, repetition=None,
+                 modelName=None):
+        """
+        Initializes the kineticsOpenSimAD class for extracting data from a
+        dynamic simulation.
+
+        Args:
+            data_dir (str): The directory where data is stored.
+            session_id (str): The identifier for the data collection session.
+            trial_name (str): The name of the motion trial.
+            repetition (int, optional): The index of the motion repetition, only
+            applicable if motion_type is 'sit_to_stand' or 'squats' 
+            case (str): The case for which to retrieve optimal results.
+            modelName (str, optional): The name of the OpenSim model used in the 
+            simulation (ignore if default).
+
+        This class is designed to extract data from a dynamic simulation 
+        conducted using OpenSimAD.
+        """
+        
+        # Load optimal results.
+        sessionDir = os.path.join(data_dir, session_id)
+        opensimDir = os.path.join(sessionDir, 'OpenSimData')
+        repetitionSuffix = ''
+        if not repetition is None:
+            repetitionSuffix = '_rep' + str(repetition)
+        resultsDir = os.path.join(opensimDir, 'Dynamics', 
+                                  trial_name + repetitionSuffix)
+        # Check if results directory exists.
+        if not os.path.exists(resultsDir):
+            raise Exception(
+                'Results directory: ' + resultsDir + ' does not exist. \
+                      Have you run the simulation for ' + trial_name + ' ? \
+                        Have you set the repetition index if simulating \
+                            a sit to stand or a squat?')
+
+        optimal_results = np.load(
+            os.path.join(resultsDir, 'optimaltrajectories.npy'), 
+            allow_pickle=True).item() 
+
+        self.optimal_result = optimal_results[case]
```

Comment:
should we make the case argument to the class optional and if there's only one case, automatically detect it?